gqlalchemy/transformations.py

```diff
@@ -50,21 +50,21 @@ def nx_graph_to_memgraph_parallel(
     password: str = "",
     encrypted: bool = False,
     create_index=False,
+    config: NetworkXCypherConfig = NetworkXCypherConfig(),
 ) -> None:
     """Generates a Cypher queries and inserts data into Memgraph in parallel"""
-    query_groups = []
-    if create_index:
-        query_groups.append(_nx_nodes_to_cypher_with_index(graph))
-    else:
+    builder = NetworkXCypherBuilder(create_index=create_index, config=config)
+    query_groups = builder.yield_query_groups(graph)
+
+    if not create_index:
```

Comment:
Not related with this PR, but could index become part of the Builder in future releases?

Reply:
I think the index part could be moved into `NetworkXCypherConfig`